<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CodeEditor</class>
 <widget class="QDialog" name="CodeEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>523</width>
    <height>710</height>
   </rect>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::NoContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>AlpacaScript Code Editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QSplitter" name="_editorSplitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QTextEdit" name="_codeEditor"/>
     <widget class="QWidget" name="verticalLayoutWidget">
      <layout class="QVBoxLayout" name="_variableTableVLayout">
       <item>
        <layout class="QHBoxLayout" name="_codeEditorLayout">
         <item>
          <widget class="QPushButton" name="_addVariable">
           <property name="toolTip">
            <string>Add a new variable entry to the variable table. This variable can be used in the alpaca script</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="resources/TACConfigEditor.qrc">
             <normaloff>:/TabsPlus.png</normaloff>:/TabsPlus.png</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="_removeVariable">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Remove an existing variable from the variable table. Select the row from the variable table and click the button</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="resources/TACConfigEditor.qrc">
             <normaloff>:/TabsRemove.png</normaloff>:/TabsRemove.png</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="_validateButton">
           <property name="toolTip">
            <string>Validate the AlpacaScript</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="resources/TACConfigEditor.qrc">
             <normaloff>:/scriptcheck.png</normaloff>:/scriptcheck.png</iconset>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableWidget" name="_variableTable">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="contextMenuPolicy">
          <enum>Qt::NoContextMenu</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::DoubleClicked|QAbstractItemView::EditKeyPressed</set>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <column>
          <property name="text">
           <string>Name</string>
          </property>
          <property name="toolTip">
           <string>The name of the variable to be added to alpaca script</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Label</string>
          </property>
          <property name="toolTip">
           <string>The label for the variable</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Tooltip</string>
          </property>
          <property name="toolTip">
           <string>The tooltip for the variable</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Type</string>
          </property>
          <property name="toolTip">
           <string>The type of variable (integer)</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Default value</string>
          </property>
          <property name="toolTip">
           <string>The value for the variable. This will be the default value in case user does not update the value from the TAC UI. The value range must be between 1000 and 60000</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Location</string>
          </property>
          <property name="toolTip">
           <string>The location of the variable input in the TAC UI</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resources/TACConfigEditor.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>CodeEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>CodeEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
